var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -659,6 +663,10 @@ def __init__(self, spec, **kwargs):
         # spec.architecture.target is like `uname -m`
         target   = spec.architecture.target
 
+        # Platform detects as 'cray' - treat like 'linux'
+        if platform == 'cray':
+            platform = 'linux'
```

Comment:
I like this approach, to just treat `cray` as a separate platform.  But I think it should be moved to an `elif` block at the bottom, and it needs to detect `self.arch_option` if we still want to use that, for consistency.

Like:
```
elif platform == 'cray':
    platform = 'linux'
    if self.arch_option == '64':
        platform += '64'
```